--- drivers/acpi/acpica/source/include/platform/acenv.h	2021-07-30 17:20:33.000000000 +0200
@@ -305,7 +305,10 @@
 
 #endif
 
-#if defined(_LINUX) || defined(__linux__)
+#if defined(__KTF__)
+#include "acktf.h"
+
+#elif defined(_LINUX) || defined(__linux__)
 #include "aclinux.h"
 
 #elif defined(_APPLE) || defined(__APPLE__)
--- drivers/acpi/acpica/source/components/namespace/nsaccess.c  2023-06-28 23:38:40.000000000 +0200
@@ -613,6 +613,8 @@
 
             if (SearchParentFlag == ACPI_NS_NO_UPSEARCH)
             {
+                /* avoid compiler error when not debugging */
+                (void)NumCarats;
                 ACPI_DEBUG_PRINT ((ACPI_DB_NAMES,
                     "Search scope is [%4.4s], path has %u carat(s)\n",
                     AcpiUtGetNodeName (ThisNode), NumCarats));