diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/Makefile b/./Makefile
index 440249f..5a32b1b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/Makefile
+++ b/./Makefile
@@ -51,6 +51,9 @@ all:
 lib:
 	$(MAKE) -C lib
 
+cleanlib:
+	@set -e; $(MAKE) -C lib clean; done
+
 clean: 
 	@set -e ; for d in $(DIRS) ; do $(MAKE) -C $$d $@ ; done
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/config.mk b/./config.mk
index cc9a6d9..5fdf575 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/config.mk
+++ b/./config.mk
@@ -28,8 +28,15 @@
 # It is included by every Makefile
 #
 #
+ifeq ($(KTF_PFMLIB_COMPILE),1)
+SYS  := KTF
+CFLAGS += -DKTF_PFMLIB_INTEGRATION
+else
 SYS  := $(shell uname -s)
+endif
 ARCH := $(shell uname -m)
+
+
 ifeq (i686,$(findstring i686,$(ARCH)))
 override ARCH=i386
 endif
@@ -92,7 +99,7 @@ endif
 # CONFIG_PFMLIB_NOPYTHON: do not generate the python support, incompatible
 # with PFMLIB_SHARED=n
 #
-CONFIG_PFMLIB_SHARED?=y
+CONFIG_PFMLIB_SHARED?=n
 CONFIG_PFMLIB_DEBUG?=y
 CONFIG_PFMLIB_NOPYTHON?=y
 
@@ -116,11 +123,14 @@ AGE=1
 #
 # Where should things (lib, headers, man) go in the end.
 #
+
+ifneq ($(KTF_PFMLIB_COMPILE),1)
 PREFIX=/usr/local
 LIBDIR=$(PREFIX)/lib
 INCDIR=$(PREFIX)/include
 MANDIR=$(PREFIX)/share/man
 DOCDIR=$(PREFIX)/share/doc/libpfm-$(VERSION).$(REVISION).$(AGE)
+endif
 
 #
 # System header files
@@ -194,6 +204,9 @@ LDCONFIG=ldconfig
 LN?=ln -sf
 PFMINCDIR=$(TOPDIR)/include
 PFMLIBDIR=$(TOPDIR)/lib
+ifeq ($(KTF_PFMLIB_COMPILE),1)
+KTF_INC_DIRS= -I$(KTF_ROOT)/include -I$(KTF_ROOT)/include/arch/x86
+endif
 #
 # -Wextra: to enable extra compiler sanity checks (e.g., signed vs. unsigned)
 # -Wno-unused-parameter: to avoid warnings on unused foo(void *this) parameter
@@ -221,6 +234,11 @@ ifeq ($(CONFIG_PFMLIB_DEBUG),y)
 CFLAGS += -DCONFIG_PFMLIB_DEBUG
 endif
 
+
+ifeq ($(KTF_PFMLIB_COMPILE),1)
+CFLAGS += $(KTF_INC_DIRS) -U__linux__ -std=gnu99 -O3 -ffreestanding -nostdlib -nostdinc
+endif
+
 CTAGS?=ctags
 
 #
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/err.h b/./include/perfmon/err.h
index 2f2c116..aca8b1d 100755
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/err.h
+++ b/./include/perfmon/err.h
@@ -22,6 +22,8 @@
 #define __PFM_ERR_H__
 
 #ifndef PFMLIB_WINDOWS
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <err.h>
 #else /* PFMLIB_WINDOWS */
 #define warnx(...) do { \
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/perf_event.h b/./include/perfmon/perf_event.h
index fcce242..4ae97cd 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/perf_event.h
+++ b/./include/perfmon/perf_event.h
@@ -24,12 +24,12 @@
 
 #pragma GCC visibility push(default)
 
-#include <sys/types.h>
-#include <unistd.h>		/* for syscall numbers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <inttypes.h>
-#include <sys/syscall.h>	/* for syscall stub macros */
-#include <sys/ioctl.h>		/* for _IO */
-#include <sys/prctl.h>		/* for prctl() comamnds */
+
+/* Manually including errno.h*/
+#include <errno.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -516,6 +516,9 @@ enum perf_callchain_context {
 /*
  * perf_event_open() syscall stub
  */
+
+/*
+ * dkgupta: No linux kernel underneath, thus there is no perf_event_open. Commeting it out
 static inline int
 perf_event_open(
 	struct perf_event_attr		*hw_event_uptr,
@@ -527,6 +530,7 @@ perf_event_open(
 	return syscall(
 		__NR_perf_event_open, hw_event_uptr, pid, cpu, group_fd, flags);
 }
+*/
 
 /*
  * compensate for some distros which do not
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/pfmlib.h b/./include/perfmon/pfmlib.h
index 0edb59c..e3bd34b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/pfmlib.h
+++ b/./include/perfmon/pfmlib.h
@@ -31,10 +31,12 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-#include <sys/types.h>
-#include <unistd.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <inttypes.h>
-#include <stdio.h>
+
+/* dkgupta: No FILE on KTF, punting to void */
+typedef void FILE;
 
 #define LIBPFM_VERSION		(4 << 16 | 0)
 #define PFM_MAJ_VERSION(v)	((v)>>16)
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/pfmlib_perf_event.h b/./include/perfmon/pfmlib_perf_event.h
index 0516277..f9ba8a5 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/include/perfmon/pfmlib_perf_event.h
+++ b/./include/perfmon/pfmlib_perf_event.h
@@ -22,6 +22,8 @@
 #ifndef __PFMLIB_PERF_EVENTS_H__
 #define __PFMLIB_PERF_EVENTS_H__
 
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <perfmon/pfmlib.h>
 #include <perfmon/perf_event.h>
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64.c b/./lib/pfmlib_amd64.c
index f093e20..483ad2b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64.c
+++ b/./lib/pfmlib_amd64.c
@@ -26,9 +26,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 /* private headers */
 #include "pfmlib_priv.h"		/* library private */
@@ -188,8 +188,19 @@ amd64_get_revision(pfm_amd64_config_t *cfg)
  * Some gcc's (4.1.2 on Core2) object to pairing push/pop and ebx in 64 bit mode.
  * Using the opcode directly avoids this problem.
  */
+
+/*
+ * cpuid name def is colliding with cpuid definition in KTF. 
+ * for sake of keeping this library self contained and using its own cpuid impl
+ * Thus changing its name
+ */
+#ifdef KTF_PFMLIB_INTEGRATION
+static inline void
+cpuid_amd(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#else
 static inline void
 cpuid(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#endif
 {
   __asm__ __volatile__ (".byte 0x53\n\tcpuid\n\tmovl %%ebx, %%esi\n\t.byte 0x5b"
        : "=a" (*a),
@@ -326,7 +337,11 @@ pfm_amd64_detect(void *this)
 	if (pfm_amd64_cfg.family)
 		return PFM_SUCCESS;
 
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_amd(0, &a, &b, &c, &d);
+#else
 	cpuid(0, &a, &b, &c, &d);
+#endif
 	strncpy(&buffer[0], (char *)(&b), 4);
 	strncpy(&buffer[4], (char *)(&d), 4);
 	strncpy(&buffer[8], (char *)(&c), 4);
@@ -335,7 +350,11 @@ pfm_amd64_detect(void *this)
 	if (strcmp(buffer, "AuthenticAMD"))
 		return PFM_ERR_NOTSUPP;
 
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_amd(1, &a, &b, &c, &d);
+#else
 	cpuid(1, &a, &b, &c, &d);
+#endif
 	pfm_amd64_cfg.family = (a >> 8) & 0x0000000f;  // bits 11 - 8
 	pfm_amd64_cfg.model  = (a >> 4) & 0x0000000f;  // Bits  7 - 4
 	if (pfm_amd64_cfg.family == 0xf) {
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam10h.c b/./lib/pfmlib_amd64_fam10h.c
index 60df01d..43cc653 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam10h.c
+++ b/./lib/pfmlib_amd64_fam10h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam10h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam11h.c b/./lib/pfmlib_amd64_fam11h.c
index 6917e9c..509589f 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam11h.c
+++ b/./lib/pfmlib_amd64_fam11h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam11h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam12h.c b/./lib/pfmlib_amd64_fam12h.c
index bd84167..e3c4d70 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam12h.c
+++ b/./lib/pfmlib_amd64_fam12h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam12h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam14h.c b/./lib/pfmlib_amd64_fam14h.c
index 2f1fcb8..0578137 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam14h.c
+++ b/./lib/pfmlib_amd64_fam14h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam14h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam15h.c b/./lib/pfmlib_amd64_fam15h.c
index 4d14f46..a4a459b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam15h.c
+++ b/./lib/pfmlib_amd64_fam15h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam15h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam16h.c b/./lib/pfmlib_amd64_fam16h.c
index 3ee1ba7..578a66d 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam16h.c
+++ b/./lib/pfmlib_amd64_fam16h.c
@@ -23,6 +23,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam16h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam17h.c b/./lib/pfmlib_amd64_fam17h.c
index 443b25f..649b4b9 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_fam17h.c
+++ b/./lib/pfmlib_amd64_fam17h.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_fam17h.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_k7.c b/./lib/pfmlib_amd64_k7.c
index 1601ad2..41e414f 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_k7.c
+++ b/./lib/pfmlib_amd64_k7.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_k7.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_k8.c b/./lib/pfmlib_amd64_k8.c
index 9c2872b..73cbddb 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_k8.c
+++ b/./lib/pfmlib_amd64_k8.c
@@ -24,6 +24,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_amd64_priv.h"
 #include "events/amd64_events_k8.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_perf_event.c b/./lib/pfmlib_amd64_perf_event.c
index e210328..d5c84d7 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_amd64_perf_event.c
+++ b/./lib/pfmlib_amd64_perf_event.c
@@ -21,10 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <limits.h>
 
 /* private headers */
 #include "pfmlib_priv.h"		/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm.c b/./lib/pfmlib_arm.c
index 91c35c6..be49af7 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm.c
+++ b/./lib/pfmlib_arm.c
@@ -23,11 +23,9 @@
  * 
  */
 
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv6.c b/./lib/pfmlib_arm_armv6.c
index d59b402..760d912 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv6.c
+++ b/./lib/pfmlib_arm_armv6.c
@@ -22,11 +22,9 @@
  *
  */
 
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv7_pmuv1.c b/./lib/pfmlib_arm_armv7_pmuv1.c
index c930781..f4e347b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv7_pmuv1.c
+++ b/./lib/pfmlib_arm_armv7_pmuv1.c
@@ -23,11 +23,9 @@
  * 
  */
 
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv8.c b/./lib/pfmlib_arm_armv8.c
index 0a3313f..5f4c596 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_armv8.c
+++ b/./lib/pfmlib_arm_armv8.c
@@ -22,9 +22,9 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  *
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_perf_event.c b/./lib/pfmlib_arm_perf_event.c
index c19df5f..c7a88c0 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_arm_perf_event.c
+++ b/./lib/pfmlib_arm_perf_event.c
@@ -21,9 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 /* private headers */
 #include "pfmlib_priv.h"		/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_cell.c b/./lib/pfmlib_cell.c
index 07f7446..d42ec26 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_cell.c
+++ b/./lib/pfmlib_cell.c
@@ -22,11 +22,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
-#include <stdlib.h>
 
 /* public headers */
 #include <perfmon/pfmlib_cell.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_common.c b/./lib/pfmlib_common.c
index ca93d72..26834b7 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_common.c
+++ b/./lib/pfmlib_common.c
@@ -25,18 +25,18 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <stdarg.h>
-#include <limits.h>
 
 #include <perfmon/pfmlib.h>
 
 #include "pfmlib_priv.h"
 
+#ifdef KTF_PFMLIB_INTEGRATION
+#include <mm/slab.h>
+#endif
+
 static pfmlib_pmu_t *pfmlib_pmus[]=
 {
 
@@ -573,6 +573,11 @@ __pfm_vbprintf(const char *fmt, ...)
  * This avoids a dependency on having a C library with
  * support for getline().
  */
+
+/*
+ * Below function is not needed by KTF and it uses realloc which is not implemented in KTF mm.
+ */
+#ifndef KTF_PFMLIB_INTEGRATION
 int
 pfmlib_getl(char **buffer, size_t *len, FILE *fp)
 {
@@ -620,7 +625,7 @@ pfmlib_getl(char **buffer, size_t *len, FILE *fp)
 	b[i] = '\0';
 	return c != EOF ? 0 : -1;
 }
-
+#endif
 
 
 /*
@@ -811,18 +816,29 @@ pfmlib_check_struct(void *st, size_t usz, size_t refsz, size_t sz)
 static void
 pfmlib_init_env(void)
 {
+#ifndef KTF_PFMLIB_INTEGRATION	
 	char *str;
 
 	pfm_cfg.fp = stderr;
+#else
+	pfm_cfg.fp = NULL;
+#endif
 
+#ifndef KTF_PFMLIB_INTEGRATION	
 	str = getenv("LIBPFM_VERBOSE");
 	if (str && isdigit((int)*str))
 		pfm_cfg.verbose = *str - '0';
+#else
+	pfm_cfg.verbose = 0;
+#endif	
 
+#ifndef KTF_PFMLIB_INTEGRATION	
 	str = getenv("LIBPFM_DEBUG");
 	if (str && isdigit((int)*str))
 		pfm_cfg.debug = *str - '0';
+#endif
 
+#ifndef KTF_PFMLIB_INTEGRATION	
 	str = getenv("LIBPFM_DEBUG_STDOUT");
 	if (str)
 		pfm_cfg.fp = stdout;
@@ -836,6 +852,7 @@ pfmlib_init_env(void)
 	str = getenv("LIBPFM_DISABLED_PMUS");
 	if (str)
 		pfm_cfg.blacklist_pmus = str;
+#endif
 }
 
 static int
@@ -861,13 +878,24 @@ pfmlib_pmu_sanity_checks(pfmlib_pmu_t *p)
 int
 pfmlib_build_fstr(pfmlib_event_desc_t *e, char **fstr)
 {
+#ifdef KTF_PFMLIB_INTEGRATION
+	int len = 0;
+#endif	
 	/* nothing to do */
 	if (!fstr)
 		return PFM_SUCCESS;
 
+#ifdef KTF_PFMLIB_INTEGRATION
+	len = strlen(e->fstr) + 2 + strlen(e->pmu->name) + 1;
+#endif
+
 	*fstr = malloc(strlen(e->fstr) + 2 + strlen(e->pmu->name) + 1);
 	if (*fstr)
+#ifdef KTF_PFMLIB_INTEGRATION
+		snprintf(*fstr, len, "%s::%s", e->pmu->name, e->fstr);
+#else		
 		sprintf(*fstr, "%s::%s", e->pmu->name, e->fstr);
+#endif
 
 	return *fstr ? PFM_SUCCESS : PFM_ERR_NOMEM;
 }
@@ -1774,7 +1802,11 @@ pfmlib_pmu_validate_encoding(pfmlib_pmu_t *pmu, FILE *fp)
 			/*
 			 * XXX: some events may require more than one umasks to encode
 			 */
+#ifdef KTF_PFMLIB_INTEGRATION	
+			snprintf(buf, maxlen, "%s::%s:%s", pmu->name, einfo.name, ainfo.name);
+#else
 			sprintf(buf, "%s::%s:%s", pmu->name, einfo.name, ainfo.name);
+#endif
 			ret = pfmlib_validate_encoding(buf, PFM_PLM3|PFM_PLM0);
 			if (ret != PFM_SUCCESS) {
 				if (pmu->can_auto_encode) {
@@ -1793,7 +1825,11 @@ pfmlib_pmu_validate_encoding(pfmlib_pmu_t *pmu, FILE *fp)
 			um++;
 		}
 		if (um == 0) {
+#ifdef KTF_PFMLIB_INTEGRATION	
+			snprintf(buf, maxlen, "%s::%s:%s", pmu->name, einfo.name, ainfo.name);
+#else
 			sprintf(buf, "%s::%s", pmu->name, einfo.name);
+#endif
 			ret = pfmlib_validate_encoding(buf, PFM_PLM3|PFM_PLM0);
 			if (ret != PFM_SUCCESS) {
 				if (pmu->can_auto_encode) {
@@ -1815,14 +1851,20 @@ pfmlib_pmu_validate_encoding(pfmlib_pmu_t *pmu, FILE *fp)
 }
 
 int
-pfm_pmu_validate(pfm_pmu_t pmu_id, FILE *fp)
+pfm_pmu_validate(pfm_pmu_t pmu_id, 
+#ifdef KTF_PFMLIB_INTEGRATION
+	       	__attribute__((unused)) 
+#endif
+		FILE *fp)
 {
 	pfmlib_pmu_t *pmu, *pmx;
 	int nos = 0;
 	int i, ret;
 
+#ifdef KTF_PFMLIB_INTEGRATION 
 	if (fp == NULL)
 		return PFM_ERR_INVAL;
+#endif
 
 	pmu = pmu2pmuidx(pmu_id);
 	if (!pmu)
@@ -2127,6 +2169,9 @@ pfmlib_get_pmu_by_type(pfm_pmu_type_t t)
 	return NULL;
 }
 
+
+#ifndef KTF_PFMLIB_INTEGRATION
+/* qsort implementaiton is punted out, so this compare routine is dead as well */
 static int
 pfmlib_compare_attr_id(const void *a, const void *b)
 {
@@ -2137,11 +2182,15 @@ pfmlib_compare_attr_id(const void *a, const void *b)
 		return -1;
 	return t1->id == t2->id ? 0 : 1;
 }
+#endif
 
 void
 pfmlib_sort_attr(pfmlib_event_desc_t *e)
 {
+#ifndef KTF_PFMLIB_INTEGRATION
+	/* dkgupta: we don't have qsort implementation, plus we dont need sorting of events now. */
 	qsort(e->attrs, e->nattrs, sizeof(pfmlib_attr_t), pfmlib_compare_attr_id);
+#endif
 }
 
 static int
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_gen_ia64.c b/./lib/pfmlib_gen_ia64.c
index 9a9789f..a1092b5 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_gen_ia64.c
+++ b/./lib/pfmlib_gen_ia64.c
@@ -21,10 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
-#include <stdlib.h>
 
 #include <perfmon/pfmlib.h>
 #include <perfmon/pfmlib_gen_ia64.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_atom.c b/./lib/pfmlib_intel_atom.c
index 5ce95a3..1c9aeea 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_atom.c
+++ b/./lib/pfmlib_intel_atom.c
@@ -33,6 +33,8 @@
  * Intel Atom = architectural v3 + PEBS
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_atom_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdw.c b/./lib/pfmlib_intel_bdw.c
index 1de8438..d50e23c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdw.c
+++ b/./lib/pfmlib_intel_bdw.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_bdw_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_cbo.c b/./lib/pfmlib_intel_bdx_unc_cbo.c
index d1ff970..fcf0d67 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_cbo.c
+++ b/./lib/pfmlib_intel_bdx_unc_cbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_ha.c b/./lib/pfmlib_intel_bdx_unc_ha.c
index d928ba0..7a1dd7c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_ha.c
+++ b/./lib/pfmlib_intel_bdx_unc_ha.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_imc.c b/./lib/pfmlib_intel_bdx_unc_imc.c
index 462f547..7f6258c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_imc.c
+++ b/./lib/pfmlib_intel_bdx_unc_imc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_irp.c b/./lib/pfmlib_intel_bdx_unc_irp.c
index 14e010b..66bf0f4 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_irp.c
+++ b/./lib/pfmlib_intel_bdx_unc_irp.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_pcu.c b/./lib/pfmlib_intel_bdx_unc_pcu.c
index f2ffe7f..c51fc7c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_pcu.c
+++ b/./lib/pfmlib_intel_bdx_unc_pcu.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_qpi.c b/./lib/pfmlib_intel_bdx_unc_qpi.c
index e3fba3d..89a81db 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_qpi.c
+++ b/./lib/pfmlib_intel_bdx_unc_qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_r2pcie.c b/./lib/pfmlib_intel_bdx_unc_r2pcie.c
index 18bed6c..e8373ac 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_r2pcie.c
+++ b/./lib/pfmlib_intel_bdx_unc_r2pcie.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_r3qpi.c b/./lib/pfmlib_intel_bdx_unc_r3qpi.c
index 89ba498..8c8451d 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_r3qpi.c
+++ b/./lib/pfmlib_intel_bdx_unc_r3qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_sbo.c b/./lib/pfmlib_intel_bdx_unc_sbo.c
index a2be8bc..f29beb4 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_sbo.c
+++ b/./lib/pfmlib_intel_bdx_unc_sbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_ubo.c b/./lib/pfmlib_intel_bdx_unc_ubo.c
index f0d058a..dbe110e 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_bdx_unc_ubo.c
+++ b/./lib/pfmlib_intel_bdx_unc_ubo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_core.c b/./lib/pfmlib_intel_core.c
index 2eb42fa..d70f9d6 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_core.c
+++ b/./lib/pfmlib_intel_core.c
@@ -29,6 +29,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_core_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_coreduo.c b/./lib/pfmlib_intel_coreduo.c
index d538bbe..65b22d0 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_coreduo.c
+++ b/./lib/pfmlib_intel_coreduo.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"			/* library private */
 #include "pfmlib_intel_x86_priv.h"		/* architecture private */
 #include "events/intel_coreduo_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_glm.c b/./lib/pfmlib_intel_glm.c
index 0b8bd9d..a746a38 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_glm.c
+++ b/./lib/pfmlib_intel_glm.c
@@ -23,6 +23,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_glm_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hsw.c b/./lib/pfmlib_intel_hsw.c
index f0c280d..37c1894 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hsw.c
+++ b/./lib/pfmlib_intel_hsw.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_hsw_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_cbo.c b/./lib/pfmlib_intel_hswep_unc_cbo.c
index f5bbeb3..5c7f3f6 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_cbo.c
+++ b/./lib/pfmlib_intel_hswep_unc_cbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_ha.c b/./lib/pfmlib_intel_hswep_unc_ha.c
index f5073bb..73fb435 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_ha.c
+++ b/./lib/pfmlib_intel_hswep_unc_ha.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_imc.c b/./lib/pfmlib_intel_hswep_unc_imc.c
index 9d7efa9..2610210 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_imc.c
+++ b/./lib/pfmlib_intel_hswep_unc_imc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_irp.c b/./lib/pfmlib_intel_hswep_unc_irp.c
index 0a390b6..43f5827 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_irp.c
+++ b/./lib/pfmlib_intel_hswep_unc_irp.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_pcu.c b/./lib/pfmlib_intel_hswep_unc_pcu.c
index 91fe13e..40da578 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_pcu.c
+++ b/./lib/pfmlib_intel_hswep_unc_pcu.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_qpi.c b/./lib/pfmlib_intel_hswep_unc_qpi.c
index 7cedcef..0df6391 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_qpi.c
+++ b/./lib/pfmlib_intel_hswep_unc_qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_r2pcie.c b/./lib/pfmlib_intel_hswep_unc_r2pcie.c
index f3f82d6..e171c47 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_r2pcie.c
+++ b/./lib/pfmlib_intel_hswep_unc_r2pcie.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_r3qpi.c b/./lib/pfmlib_intel_hswep_unc_r3qpi.c
index 8861901..2dcef9b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_r3qpi.c
+++ b/./lib/pfmlib_intel_hswep_unc_r3qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_sbo.c b/./lib/pfmlib_intel_hswep_unc_sbo.c
index e99f599..2710419 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_sbo.c
+++ b/./lib/pfmlib_intel_hswep_unc_sbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_ubo.c b/./lib/pfmlib_intel_hswep_unc_ubo.c
index b161b25..7526a2f 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_hswep_unc_ubo.c
+++ b/./lib/pfmlib_intel_hswep_unc_ubo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivb.c b/./lib/pfmlib_intel_ivb.c
index 696c88b..75b8668 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivb.c
+++ b/./lib/pfmlib_intel_ivb.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_ivb_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivb_unc.c b/./lib/pfmlib_intel_ivb_unc.c
index d3160d2..d9e70ad 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivb_unc.c
+++ b/./lib/pfmlib_intel_ivb_unc.c
@@ -21,6 +21,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_cbo.c b/./lib/pfmlib_intel_ivbep_unc_cbo.c
index 6053aa5..70f3ce7 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_cbo.c
+++ b/./lib/pfmlib_intel_ivbep_unc_cbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_ha.c b/./lib/pfmlib_intel_ivbep_unc_ha.c
index d4bd6f9..a10db13 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_ha.c
+++ b/./lib/pfmlib_intel_ivbep_unc_ha.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_imc.c b/./lib/pfmlib_intel_ivbep_unc_imc.c
index aa7d3a8..023a25c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_imc.c
+++ b/./lib/pfmlib_intel_ivbep_unc_imc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_irp.c b/./lib/pfmlib_intel_ivbep_unc_irp.c
index 84fc22f..712cd9b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_irp.c
+++ b/./lib/pfmlib_intel_ivbep_unc_irp.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_pcu.c b/./lib/pfmlib_intel_ivbep_unc_pcu.c
index 222ed45..51c6333 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_pcu.c
+++ b/./lib/pfmlib_intel_ivbep_unc_pcu.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_qpi.c b/./lib/pfmlib_intel_ivbep_unc_qpi.c
index e68eb75..59e1b0b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_qpi.c
+++ b/./lib/pfmlib_intel_ivbep_unc_qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_r2pcie.c b/./lib/pfmlib_intel_ivbep_unc_r2pcie.c
index 87c680c..6ace068 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_r2pcie.c
+++ b/./lib/pfmlib_intel_ivbep_unc_r2pcie.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_r3qpi.c b/./lib/pfmlib_intel_ivbep_unc_r3qpi.c
index 5ed7e4d..bbd8e24 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_r3qpi.c
+++ b/./lib/pfmlib_intel_ivbep_unc_r3qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_ubo.c b/./lib/pfmlib_intel_ivbep_unc_ubo.c
index db7f629..06e2d19 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_ivbep_unc_ubo.c
+++ b/./lib/pfmlib_intel_ivbep_unc_ubo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knc.c b/./lib/pfmlib_intel_knc.c
index e65667e..7a348da 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knc.c
+++ b/./lib/pfmlib_intel_knc.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"			/* library private */
 #include "pfmlib_intel_x86_priv.h"		/* architecture private */
 #include "events/intel_knc_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl.c b/./lib/pfmlib_intel_knl.c
index 5cfd545..d93486b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl.c
+++ b/./lib/pfmlib_intel_knl.c
@@ -29,6 +29,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_knl_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_cha.c b/./lib/pfmlib_intel_knl_unc_cha.c
index 6ea446c..b55c7d9 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_cha.c
+++ b/./lib/pfmlib_intel_knl_unc_cha.c
@@ -25,11 +25,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_edc.c b/./lib/pfmlib_intel_knl_unc_edc.c
index 1e28638..0d4c8ba 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_edc.c
+++ b/./lib/pfmlib_intel_knl_unc_edc.c
@@ -25,11 +25,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_imc.c b/./lib/pfmlib_intel_knl_unc_imc.c
index 02280f3..27b6737 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_imc.c
+++ b/./lib/pfmlib_intel_knl_unc_imc.c
@@ -25,11 +25,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_m2pcie.c b/./lib/pfmlib_intel_knl_unc_m2pcie.c
index fafc6ec..bf9ac5c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_knl_unc_m2pcie.c
+++ b/./lib/pfmlib_intel_knl_unc_m2pcie.c
@@ -25,11 +25,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_netburst.c b/./lib/pfmlib_intel_netburst.c
index 9b49605..1b092cc 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_netburst.c
+++ b/./lib/pfmlib_intel_netburst.c
@@ -26,6 +26,8 @@
  * Support for the Pentium4/Xeon/EM64T processor family (family=15).
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_netburst_priv.h"
 #include "pfmlib_intel_x86_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_netburst_perf_event.c b/./lib/pfmlib_intel_netburst_perf_event.c
index a2f32eb..4f259c5 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_netburst_perf_event.c
+++ b/./lib/pfmlib_intel_netburst_perf_event.c
@@ -22,11 +22,9 @@
  *
  * This file implements the common code for all Intel X86 processors.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_nhm.c b/./lib/pfmlib_intel_nhm.c
index 87ce5cf..a1bd588 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_nhm.c
+++ b/./lib/pfmlib_intel_nhm.c
@@ -24,6 +24,8 @@
  * Nehalem PMU = architectural perfmon v3 + OFFCORE + PEBS v2 + LBR
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_nhm_unc.c b/./lib/pfmlib_intel_nhm_unc.c
index e9bae3d..d286c22 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_nhm_unc.c
+++ b/./lib/pfmlib_intel_nhm_unc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_p6.c b/./lib/pfmlib_intel_p6.c
index fa33874..5062746 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_p6.c
+++ b/./lib/pfmlib_intel_p6.c
@@ -23,6 +23,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"			/* library private */
 #include "pfmlib_intel_x86_priv.h"		/* architecture private */
 #include "events/intel_p6_events.h"		/* generic P6 (PIII) */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_rapl.c b/./lib/pfmlib_intel_rapl.c
index 8a04079..f38cf16 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_rapl.c
+++ b/./lib/pfmlib_intel_rapl.c
@@ -29,6 +29,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 /*
  * for now, we reuse the x86 table entry format and callback to avoid duplicating
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skl.c b/./lib/pfmlib_intel_skl.c
index 1754310..e76da93 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skl.c
+++ b/./lib/pfmlib_intel_skl.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_skl_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_cha.c b/./lib/pfmlib_intel_skx_unc_cha.c
index c5f8141..0dc6786 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_cha.c
+++ b/./lib/pfmlib_intel_skx_unc_cha.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_iio.c b/./lib/pfmlib_intel_skx_unc_iio.c
index ad6ca88..9ef5bac 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_iio.c
+++ b/./lib/pfmlib_intel_skx_unc_iio.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_imc.c b/./lib/pfmlib_intel_skx_unc_imc.c
index 8dea562..965df6c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_imc.c
+++ b/./lib/pfmlib_intel_skx_unc_imc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_irp.c b/./lib/pfmlib_intel_skx_unc_irp.c
index efebe46..b92f84d 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_irp.c
+++ b/./lib/pfmlib_intel_skx_unc_irp.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_m2m.c b/./lib/pfmlib_intel_skx_unc_m2m.c
index effc642..532ab99 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_m2m.c
+++ b/./lib/pfmlib_intel_skx_unc_m2m.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_m3upi.c b/./lib/pfmlib_intel_skx_unc_m3upi.c
index 97e54c3..d06b54b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_m3upi.c
+++ b/./lib/pfmlib_intel_skx_unc_m3upi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_pcu.c b/./lib/pfmlib_intel_skx_unc_pcu.c
index 5607479..6f02783 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_pcu.c
+++ b/./lib/pfmlib_intel_skx_unc_pcu.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_ubo.c b/./lib/pfmlib_intel_skx_unc_ubo.c
index eaa4edb..a053976 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_ubo.c
+++ b/./lib/pfmlib_intel_skx_unc_ubo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_upi.c b/./lib/pfmlib_intel_skx_unc_upi.c
index c2345f6..5f1b908 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_skx_unc_upi.c
+++ b/./lib/pfmlib_intel_skx_unc_upi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_slm.c b/./lib/pfmlib_intel_slm.c
index bf2560d..edb0a26 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_slm.c
+++ b/./lib/pfmlib_intel_slm.c
@@ -25,6 +25,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_slm_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snb.c b/./lib/pfmlib_intel_snb.c
index b458994..b34fb3e 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snb.c
+++ b/./lib/pfmlib_intel_snb.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_snb_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snb_unc.c b/./lib/pfmlib_intel_snb_unc.c
index edae361..de189dc 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snb_unc.c
+++ b/./lib/pfmlib_intel_snb_unc.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc.c b/./lib/pfmlib_intel_snbep_unc.c
index d4adc51..de6b149 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc.c
+++ b/./lib/pfmlib_intel_snbep_unc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_cbo.c b/./lib/pfmlib_intel_snbep_unc_cbo.c
index b1dd1d0..4db2b1c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_cbo.c
+++ b/./lib/pfmlib_intel_snbep_unc_cbo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_ha.c b/./lib/pfmlib_intel_snbep_unc_ha.c
index 28e6b71..a20f8e3 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_ha.c
+++ b/./lib/pfmlib_intel_snbep_unc_ha.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_imc.c b/./lib/pfmlib_intel_snbep_unc_imc.c
index c0cc0e1..7c86a26 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_imc.c
+++ b/./lib/pfmlib_intel_snbep_unc_imc.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_pcu.c b/./lib/pfmlib_intel_snbep_unc_pcu.c
index 5d4dd47..759222e 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_pcu.c
+++ b/./lib/pfmlib_intel_snbep_unc_pcu.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_perf_event.c b/./lib/pfmlib_intel_snbep_unc_perf_event.c
index c8ce0cb..48515dd 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_perf_event.c
+++ b/./lib/pfmlib_intel_snbep_unc_perf_event.c
@@ -20,12 +20,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
-#include <limits.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_qpi.c b/./lib/pfmlib_intel_snbep_unc_qpi.c
index 9cc4bac..04b56f1 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_qpi.c
+++ b/./lib/pfmlib_intel_snbep_unc_qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_r2pcie.c b/./lib/pfmlib_intel_snbep_unc_r2pcie.c
index c6a64b9..0c11345 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_r2pcie.c
+++ b/./lib/pfmlib_intel_snbep_unc_r2pcie.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_r3qpi.c b/./lib/pfmlib_intel_snbep_unc_r3qpi.c
index 28bb83a..81a4604 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_r3qpi.c
+++ b/./lib/pfmlib_intel_snbep_unc_r3qpi.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_ubo.c b/./lib/pfmlib_intel_snbep_unc_ubo.c
index 3b8abff..d032a54 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_snbep_unc_ubo.c
+++ b/./lib/pfmlib_intel_snbep_unc_ubo.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_wsm.c b/./lib/pfmlib_intel_wsm.c
index 2b581b4..46a1ad8 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_wsm.c
+++ b/./lib/pfmlib_intel_wsm.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_intel_x86_priv.h"
 #include "events/intel_wsm_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86.c b/./lib/pfmlib_intel_x86.c
index 34a3148..442f2b6 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86.c
+++ b/./lib/pfmlib_intel_x86.c
@@ -22,11 +22,9 @@
  *
  * This file implements the common code for all Intel X86 processors.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
@@ -56,14 +54,25 @@ pfm_intel_x86_config_t pfm_intel_x86_cfg;
  * Some gcc's (4.1.2 on Core2) object to pairing push/pop and ebx in 64 bit mode.
  * Using the opcode directly avoids this problem.
  */
+
+/*
+ * cpuid name def is colliding with cpuid definition in KTF.
+ * for sake of keeping this library self contained and using its own cpuid impl
+ * Thus changing its name
+ */
+#ifdef KTF_PFMLIB_INTEGRATION
+static inline void
+cpuid_intel(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#else
 static inline void
 cpuid(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#endif
 {
   __asm__ __volatile__ (".byte 0x53\n\tcpuid\n\tmovl %%ebx, %%esi\n\t.byte 0x5b"
        : "=a" (*a),
-	     "=S" (*b),
-		 "=c" (*c),
-		 "=d" (*d)
+	 "=S" (*b),
+	 "=c" (*c),
+	 "=d" (*d)
        : "a" (op));
 }
 
@@ -149,7 +158,11 @@ pfm_intel_x86_detect(void)
 	if (pfm_intel_x86_cfg.family)
 		return PFM_SUCCESS;
 
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_intel(0, &a, &b, &c, &d);
+#else
 	cpuid(0, &a, &b, &c, &d);
+#endif
 	strncpy(&buffer[0], (char *)(&b), 4);
 	strncpy(&buffer[4], (char *)(&d), 4);
 	strncpy(&buffer[8], (char *)(&c), 4);
@@ -159,7 +172,11 @@ pfm_intel_x86_detect(void)
 	if (strcmp(buffer, "GenuineIntel"))
 		return PFM_ERR_NOTSUPP;
 
-	cpuid(1, &a, &b, &c, &d);
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_intel(0, &a, &b, &c, &d);
+#else
+	cpuid(0, &a, &b, &c, &d);
+#endif
 
 	pfm_intel_x86_cfg.family = (a >> 8) & 0xf;  // bits 11 - 8
 	pfm_intel_x86_cfg.model  = (a >> 4) & 0xf;  // Bits  7 - 4
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86_arch.c b/./lib/pfmlib_intel_x86_arch.c
index e3af3f4..f7a6062 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86_arch.c
+++ b/./lib/pfmlib_intel_x86_arch.c
@@ -28,11 +28,9 @@
  * 	"IA-32 Intel Architecture Software Developer's Manual - Volume 3B: System
  * 	Programming Guide"
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
@@ -49,14 +47,25 @@ static intel_x86_entry_t *x86_arch_pe;
  * Some gcc's (4.1.2 on Core2) object to pairing push/pop and ebx in 64 bit mode.
  * Using the opcode directly avoids this problem.
  */
+
+/*
+ * cpuid name def is colliding with cpuid definition in KTF.
+ * for sake of keeping this library self contained and using its own cpuid impl
+ * Thus changing its name
+ */
+#ifdef KTF_PFMLIB_INTEGRATION
+static inline void
+cpuid_intel(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#else
 static inline void
 cpuid(unsigned int op, unsigned int *a, unsigned int *b, unsigned int *c, unsigned int *d)
+#endif
 {
   __asm__ __volatile__ (".byte 0x53\n\tcpuid\n\tmovl %%ebx, %%esi\n\t.byte 0x5b"
        : "=a" (*a),
-	     "=S" (*b),
-		 "=c" (*c),
-		 "=d" (*d)
+	 "=S" (*b),
+	 "=c" (*c),
+	 "=d" (*d)
        : "a" (op));
 }
 
@@ -83,7 +92,11 @@ create_arch_event_table(unsigned int mask, int version)
 	}
 	intel_x86_arch_support.pme_count = num_events;
 
+#ifdef KTF_PFMLIB_INTEGRATION
+	pe = malloc(sizeof(intel_x86_entry_t)*num_events);
+#else
 	pe = calloc(num_events, sizeof(intel_x86_entry_t));
+#endif
 	if (pe == NULL)
 		return PFM_ERR_NOTSUPP;
 
@@ -137,14 +150,22 @@ check_arch_pmu(int family)
 	 * 0xa started with Core Duo. Needed to detect if
 	 * architected PMU is present
 	 */
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_intel(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#else
 	cpuid(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#endif
 	if (eax.val < 0xa)
 		return PFM_ERR_NOTSUPP;
 
 	/*
 	 * extract architected PMU information
 	 */
-	cpuid(0xa, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_intel(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#else
+	cpuid(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#endif
 
 	/*
 	 * version must be greater than zero
@@ -177,7 +198,11 @@ pfm_intel_x86_arch_init(void *this)
 	 * extract architected PMU information
 	 */
 	if (!pfm_cfg.forced_pmu) {
-		cpuid(0xa, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#ifdef KTF_PFMLIB_INTEGRATION
+	cpuid_intel(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#else
+	cpuid(0x0, &eax.val, &ebx.val, &ecx.val, &edx.val);
+#endif
 		intel_x86_arch_support.num_cntrs = eax.eax.num_cnt;
 		intel_x86_arch_support.num_fixed_cntrs = edx.edx.num_cnt;
 	} else {
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86_perf_event.c b/./lib/pfmlib_intel_x86_perf_event.c
index 83c0294..e9e7279 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_intel_x86_perf_event.c
+++ b/./lib/pfmlib_intel_x86_perf_event.c
@@ -20,12 +20,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
-#include <limits.h>
 
 /* private headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_itanium.c b/./lib/pfmlib_itanium.c
index 28e245a..d020f5a 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_itanium.c
+++ b/./lib/pfmlib_itanium.c
@@ -21,10 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
 
 /* public headers */
 #include <perfmon/pfmlib_itanium.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_itanium2.c b/./lib/pfmlib_itanium2.c
index 0ecd123..463c5d2 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_itanium2.c
+++ b/./lib/pfmlib_itanium2.c
@@ -21,10 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
 
 /* public headers */
 #include <perfmon/pfmlib_itanium2.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips.c b/./lib/pfmlib_mips.c
index 61db613..5945159 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips.c
+++ b/./lib/pfmlib_mips.c
@@ -23,11 +23,9 @@
  *
  */
 
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips_74k.c b/./lib/pfmlib_mips_74k.c
index 6620f62..0696e50 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips_74k.c
+++ b/./lib/pfmlib_mips_74k.c
@@ -22,7 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  *
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
 #include "pfmlib_priv.h"		/* library private */
 #include "pfmlib_mips_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips_perf_event.c b/./lib/pfmlib_mips_perf_event.c
index 0c3b442..907470e 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_mips_perf_event.c
+++ b/./lib/pfmlib_mips_perf_event.c
@@ -21,9 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 #include "pfmlib_priv.h"
 #include "pfmlib_mips_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_montecito.c b/./lib/pfmlib_montecito.c
index 88fa0b3..edda1ab 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_montecito.c
+++ b/./lib/pfmlib_montecito.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
-#include <stdlib.h>
 
 /* public headers */
 #include <perfmon/pfmlib_montecito.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event.c b/./lib/pfmlib_perf_event.c
index 6a56af5..6dc95d0 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event.c
+++ b/./lib/pfmlib_perf_event.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <unistd.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <limits.h>
 #include <perfmon/pfmlib_perf_event.h>
 
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_pmu.c b/./lib/pfmlib_perf_event_pmu.c
index 6f890a9..ce93e66 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_pmu.c
+++ b/./lib/pfmlib_perf_event_pmu.c
@@ -21,16 +21,12 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <unistd.h>
 #include <dirent.h>
 #include <fcntl.h>
-#include <limits.h>
 #ifdef __linux__
-#include <sys/syscall.h> /* for openat() */
 #include <sys/param.h>
 #endif
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_priv.h b/./lib/pfmlib_perf_event_priv.h
index 31e9d0f..eac7635 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_priv.h
+++ b/./lib/pfmlib_perf_event_priv.h
@@ -23,6 +23,8 @@
  */
 #ifndef __PERF_EVENT_PRIV_H__
 #define __PERF_EVENT_PRIV_H__
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "perfmon/pfmlib_perf_event.h"
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_raw.c b/./lib/pfmlib_perf_event_raw.c
index 71d9443..5c33bb8 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_perf_event_raw.c
+++ b/./lib/pfmlib_perf_event_raw.c
@@ -21,12 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <unistd.h>
-#include <ctype.h>
 
 #include "pfmlib_priv.h"
 #include "pfmlib_perf_event_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power4.c b/./lib/pfmlib_power4.c
index 238e978..0ddf6ea 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power4.c
+++ b/./lib/pfmlib_power4.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power4_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power5.c b/./lib/pfmlib_power5.c
index cea2981..5aef5bd 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power5.c
+++ b/./lib/pfmlib_power5.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power5_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power6.c b/./lib/pfmlib_power6.c
index 7df06cd..006883c 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power6.c
+++ b/./lib/pfmlib_power6.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power6_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power7.c b/./lib/pfmlib_power7.c
index a32977c..5779c5a 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power7.c
+++ b/./lib/pfmlib_power7.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power7_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power8.c b/./lib/pfmlib_power8.c
index d30f036..07e9592 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power8.c
+++ b/./lib/pfmlib_power8.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power8_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power9.c b/./lib/pfmlib_power9.c
index b3807da..1140fd5 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_power9.c
+++ b/./lib/pfmlib_power9.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/power9_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc.c b/./lib/pfmlib_powerpc.c
index f32080d..e154bac 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc.c
+++ b/./lib/pfmlib_powerpc.c
@@ -25,7 +25,8 @@
  * Support for libpfm4 for the PowerPC 970, 970MP, Power4,4+,5,5+,6,7 processors.
  */
 
-#include <stdlib.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
 
 
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc_nest.c b/./lib/pfmlib_powerpc_nest.c
index 8a8bf5b..d435f98 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc_nest.c
+++ b/./lib/pfmlib_powerpc_nest.c
@@ -2,6 +2,8 @@
  * pfmlib_powerpc_nest.c
  */
 
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/powerpc_nest_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc_perf_event.c b/./lib/pfmlib_powerpc_perf_event.c
index 642439b..6ca18ec 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_powerpc_perf_event.c
+++ b/./lib/pfmlib_powerpc_perf_event.c
@@ -21,10 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <limits.h>
 
 /* private headers */
 #include "pfmlib_priv.h"		/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_ppc970.c b/./lib/pfmlib_ppc970.c
index da3d8d6..787c90b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_ppc970.c
+++ b/./lib/pfmlib_ppc970.c
@@ -22,6 +22,8 @@
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_power_priv.h"
 #include "events/ppc970_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_priv.h b/./lib/pfmlib_priv.h
index c1d874f..74eda46 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_priv.h
+++ b/./lib/pfmlib_priv.h
@@ -24,6 +24,8 @@
  */
 #ifndef __PFMLIB_PRIV_H__
 #define __PFMLIB_PRIV_H__
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <perfmon/pfmlib.h>
 #include <string.h>
 
@@ -738,4 +740,18 @@ is_empty_attr(const pfmlib_attr_desc_t *a)
 	return !a || !a->name || strlen(a->name) == 0 ? 1 : 0;
 }
 
+#ifdef KTF_PFMLIB_INTEGRATION
+#define malloc kmalloc
+#define free kfree
+#define vfprintf(fp, fmt, ...) 			\
+	do {					\
+		printk(fmt, ##__VA_ARGS__);	\
+	} while(0)
+#define strtoull strtoul
+#define	fprintf(fp, fmt, ...)			\
+	do {					\
+		printk(fmt, ##__VA_ARGS__);	\
+	} while(0)
+#endif
+
 #endif /* __PFMLIB_PRIV_H__ */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_s390x_cpumf.c b/./lib/pfmlib_s390x_cpumf.c
index 4e03fc4..d1ca33b 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_s390x_cpumf.c
+++ b/./lib/pfmlib_s390x_cpumf.c
@@ -21,11 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <ctype.h>
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <unistd.h>
 
 /* private library and arch headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_s390x_perf_event.c b/./lib/pfmlib_s390x_perf_event.c
index fb75892..e50b2ff 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_s390x_perf_event.c
+++ b/./lib/pfmlib_s390x_perf_event.c
@@ -21,9 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 /* private library and arch headers */
 #include "pfmlib_priv.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sicortex.c b/./lib/pfmlib_sicortex.c
index b9d2cad..0f099a7 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sicortex.c
+++ b/./lib/pfmlib_sicortex.c
@@ -22,12 +22,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
-#include <ctype.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdio.h>
-#include <unistd.h>
-#include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <malloc.h>
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sicortex_priv.h b/./lib/pfmlib_sicortex_priv.h
index 809fae1..7b87909 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sicortex_priv.h
+++ b/./lib/pfmlib_sicortex_priv.h
@@ -25,6 +25,8 @@
  */
 #ifndef __PFMLIB_SICORTEX_PRIV_H__
 #define __PFMLIB_SICORTEX_PRIV_H__
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_gen_mips64_priv.h"
 
 #define PFMLIB_SICORTEX_MAX_UMASK 5
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc.c b/./lib/pfmlib_sparc.c
index fe8da06..807a046 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc.c
+++ b/./lib/pfmlib_sparc.c
@@ -23,11 +23,9 @@
  * IN THE SOFTWARE.
  *
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <stdarg.h>
 
 /* private headers */
 #include "pfmlib_priv.h"			/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_niagara.c b/./lib/pfmlib_sparc_niagara.c
index d0a90ba..ad934d3 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_niagara.c
+++ b/./lib/pfmlib_sparc_niagara.c
@@ -25,6 +25,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_sparc_priv.h"
 #include "events/sparc_niagara1_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_perf_event.c b/./lib/pfmlib_sparc_perf_event.c
index 93c4b83..62f5b25 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_perf_event.c
+++ b/./lib/pfmlib_sparc_perf_event.c
@@ -21,9 +21,9 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <string.h>
-#include <stdlib.h>
 
 /* private headers */
 #include "pfmlib_priv.h"		/* library private */
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra12.c b/./lib/pfmlib_sparc_ultra12.c
index 06afaaa..8798268 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra12.c
+++ b/./lib/pfmlib_sparc_ultra12.c
@@ -25,6 +25,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_sparc_priv.h"
 #include "events/sparc_ultra12_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra3.c b/./lib/pfmlib_sparc_ultra3.c
index 7af936a..1299713 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra3.c
+++ b/./lib/pfmlib_sparc_ultra3.c
@@ -25,6 +25,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_sparc_priv.h"
 #include "events/sparc_ultra3_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra4.c b/./lib/pfmlib_sparc_ultra4.c
index 5d9e9bb..17fdb44 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_sparc_ultra4.c
+++ b/./lib/pfmlib_sparc_ultra4.c
@@ -23,6 +23,8 @@
  */
 
 /* private headers */
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include "pfmlib_priv.h"
 #include "pfmlib_sparc_priv.h"
 #include "events/sparc_ultra4plus_events.h"
diff --git a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_torrent.c b/./lib/pfmlib_torrent.c
index 72991e7..96bb02a 100644
--- a/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/libpfm_fresh/libpfm-4.10.1/lib/pfmlib_torrent.c
+++ b/./lib/pfmlib_torrent.c
@@ -21,7 +21,8 @@
  * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
  * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
-#include <sys/types.h>
+/* Auto generated disclaimer by KTF project to patch headers */
+#include <ktf.h>
 #include <dirent.h>
 #include <string.h>
 
